src/index.ts

```diff
@@ -316,10 +316,35 @@ or make sure that you already have Serverless 3.x installed in your project.
 
     const funcs = this.functions;
     const promises = [];
+    const shouldUseProviderLayers =
+      (!exclude || _.isEmpty(exclude)) &&
+      (!include ||
+        _.isEmpty(include) ||
+        (include && _.isArray(include) && include.length === funcs.length));
+    // if it's not excluding functions and (not including or including all functions)
+    if (shouldUseProviderLayers) {
+      // use providers for the layers
+      const runtime = _.get(this.serverless.service, "provider.runtime");
+      const architecture = _.get(
+        this.serverless.service,
+        "provider.architecture",
+        null
+      );
+
+      const layerArn = this.config.layerArn
+        ? this.config.layerArn
+        : await this.getLayerArn(runtime, architecture);
+
+      if (!this.serverless.service.layers) {
+        this.serverless.service.layers = {};
+      }
+      // TODO: Figure out the correct syntax to add a layer with ARN
+      this.serverless.service.layers.nrlayer = { name: layerArn };
```

Comment:
This currently doesn't work and needs the correct syntax

I have used this syntax before but it doesn't work on your plugin https://github.com/agutoli/serverless-layers/pull/89/files#diff-bfe9874d239014961b1ae4e89875a6155667db834a410aaaa2ebe3cf89820556R390

p.s. that plugin is working with my changes

Reply:
This has been fixed in my latest commit